core/file/json_utils.cpp

```diff
@@ -295,23 +288,15 @@ namespace MR
         if (pe_scheme.rows()) {
           // Assume that image being written to disk is going to have its transform adjusted,
           //   so modify the phase encoding scheme appropriately before writing to JSON
-          for (ssize_t row = 0; row != pe_scheme.rows(); ++row) {
-            Eigen::VectorXd new_line = pe_scheme.row (row);
-            for (ssize_t axis = 0; axis != 3; ++axis)
-              new_line[axis] = pe_scheme(row, order[axis]) && header.stride (order[axis]) < 0 ?
-                               -pe_scheme(row, order[axis]) :
-                               pe_scheme(row, order[axis]);
-            pe_scheme.row (row) = new_line;
-          }
-          PhaseEncoding::set_scheme (H_adj, pe_scheme);
+          PhaseEncoding::set_scheme (H_adj, PhaseEncoding::transform_for_nifti_write (pe_scheme, header));
           INFO ("Phase encoding information written to JSON file modified according to expected output NIfTI header transform realignment");
         }
         auto slice_encoding_it = H_adj.keyval().find ("SliceEncodingDirection");
         if (slice_encoding_it != H_adj.keyval().end()) {
           const Eigen::Vector3 orig_dir (Axes::id2dir (slice_encoding_it->second));
           Eigen::Vector3 new_dir;
           for (size_t axis = 0; axis != 3; ++axis)
-            new_dir[axis] = header.stride (order[axis]) > 0 ? orig_dir[order[axis]] : -orig_dir[order[axis]];
+            new_dir[axis] = flip[axis] ? orig_dir[order[axis]] : -orig_dir[order[axis]];
```

Comment:
Note that this is not just a syntax change, it changes whether the binary flipping is considered to index axes after vs. before permutation. I'll hopefully confirm for phase encoding direction once I've done a comprehensive multi-parametric test. I'm not sure I have sufficient information in my test data to actually validate this for the direction of slice encoding, but it should just follow the same logic as that of the phase encoding table.